### PR TITLE
ENH: Let consuming projects build with clang-tidy.

### PR DESCRIPTION
**Pull request by vibraphone**
When built with an environment variable (`ITK_ACCEPT_ALL_COMPILERS`) set,
pass arguments to the compiler version-check cmake utility that allow unknown
compilers. This will allow clang-tidy to be run on projects that consume ITK.

## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [x] Added test (or behavior not changed)
- [x] Updated API documentation (or API not changed)
- [x] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [x] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [x] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by vibraphone** [comment]
@thewtex @dzenanz Can you please review or assign to a reviewer? Thanks!

**Comment by vibraphone** [comment]
@hjmjohnson Friendly ping?

**Comment by dzenanz** [comment]
If Hans doesn't review today, go ahead and merge.

**Review by dzenanz — APPROVED** [review]

File: Modules/Core/Common/CMake/itkGenerateNewCompilerFeatureDetectionHeaders.cmake [hunk 1]
```diff
@@ -1,12 +1,20 @@
 ## Dynamically create compiler detection header
 include(WriteCompilerDetectionHeader)
+if (DEFINED ENV{ITK_ACCEPT_ALL_COMPILERS})
```
Comment (hjmjohnson) [inline Modules/Core/Common/CMake/itkGenerateNewCompilerFeatureDetectionHeaders.cmake:3]: This looks good.  Perhaps consider allowing both environmental variables or cmake variables to trigger the same behavior.
```
if (DEFINED ENV{ITK_ACCEPT_ALL_COMPILERS} OR ITK_ACCEPT_ALL_COMPILERS)
```

**Review by hjmjohnson — CHANGES_REQUESTED** [review]
Minor change requested to allow either environmental variable or cmake variable.

File: Modules/Core/Common/CMake/itkGenerateNewCompilerFeatureDetectionHeaders.cmake (hunk 1 above)
Comment (vibraphone) [reply Modules/Core/Common/CMake/itkGenerateNewCompilerFeatureDetectionHeaders.cmake]: That's a good idea. Done.

**Review by hjmjohnson — APPROVED** [review]

**Comment by vibraphone** [comment]
@hjmjohnson @dzenanz Thanks for the reviews. Can one of you please merge?

**Comment by dzenanz** [comment]
Once the CI is finished and all green, and there is an approval by one of the maintainers, the PR author should be able to merge too. But since you asked I will merge.

**Pull request merged**